lib/kitchen/driver/openstack.rb

```diff
@@ -303,20 +302,25 @@ def get_public_private_ips(server)
       end
 
       def get_ip(server)
-        unless config[:floating_ip].nil?
-          debug "Using floating ip: #{config[:floating_ip]}"
-          return config[:floating_ip]
-        end
-        if config[:openstack_network_name]
-          debug "Using configured net: #{config[:openstack_network_name]}"
-          return server.addresses[config[:openstack_network_name]].first['addr']
+        if config[:private_network_only] && config[:openstack_network_name]
+          update_server = compute.get_server_details(server[:server_id])  # refresh the server object with the stuff from the cloud.
+          pub = update_server[:body]['server']['addresses'][config[:openstack_network_name]][0]['addr']
+        else
+          unless config[:floating_ip].nil?
+            debug "Using floating ip: #{config[:floating_ip]}"
+            return config[:floating_ip]
+          end
+          if config[:openstack_network_name]
+            debug "Using configured net: #{config[:openstack_network_name]}"
+            return server.addresses[config[:openstack_network_name]].first['addr']
+          end
+          pub, priv = get_public_private_ips(server)
+          priv ||= server.ip_addresses unless pub
```

Comment:
hmm. I wasn't expecting both. I'd expect to have access to `server` since that's the name of the argument, but I think the value is actually the hash known as `state` elsewhere vs the Fog compute object assigned to `server` in `#create`.

The problem is that line 306 treats `server` like a hash and this line treats it like a Fog compute object.

I'll have to double check with pry tomorrow, but tracing through the code:

* `#create` receives `state` as an argument [L89](https://github.com/test-kitchen/kitchen-openstack/blob/v2.0.0/lib/kitchen/driver/openstack.rb#L89)
* `#create` defines `server`  [L96](https://github.com/test-kitchen/kitchen-openstack/blob/v2.0.0/lib/kitchen/driver/openstack.rb#L96)
* `#create` passes the `state` hash to `#wait_for_server` [L105](https://github.com/test-kitchen/kitchen-openstack/blob/v2.0.0/lib/kitchen/driver/openstack.rb#L105)
* `#wait_for_server` passes the `state` hash to `#get_ip` [L384](https://github.com/test-kitchen/kitchen-openstack/blob/v2.0.0/lib/kitchen/driver/openstack.rb#L384)
* `#get_ip` names its first argument `server` [L305](https://github.com/test-kitchen/kitchen-openstack/blob/v2.0.0/lib/kitchen/driver/openstack.rb#L305)